src/core/raster/qgssinglecolorrenderer.h

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+                         qgssinglecolorrenderer.h
+                         ---------------------------
+    begin                : April 2024
+    copyright            : (C) 2024 by Mathieu Pellerin
+    email                : mathieu at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSINGLECOLORRENDERER_H
+#define QGSSINGLECOLORRENDERER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgsrasterrenderer.h"
+
+#include <memory>
+
+class QDomElement;
+
+/**
+ * \ingroup core
+  * \brief Raster single color renderer pipe.
+  * \since QGIS 3.38
+  */
+class CORE_EXPORT QgsSingleColorRenderer: public QgsRasterRenderer
```

Comment:
```suggestion
class CORE_EXPORT QgsRasterSingleColorRenderer: public QgsRasterRenderer
```

Just to differentiate from eg a point cloud single color renderer. There's precedent here with QgsRasterContourRenderer.